flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SubpartitionTestBase.java

```diff
@@ -36,10 +38,24 @@
 	/**
 	 * Return the subpartition to be tested.
 	 */
-	abstract ResultSubpartition createSubpartition();
+	abstract ResultSubpartition createSubpartition() throws Exception;
 
 	// ------------------------------------------------------------------------
 
+	@Test
+	public void createReaderAfterDispose() throws Exception {
+		final ResultSubpartition subpartition = createSubpartition();
+		subpartition.release();
+
+		try {
+			subpartition.createReadView(() -> {});
+			fail("expected an exception");
+		}
+		catch (IllegalStateException e) {
+			// expected
```

Comment:
nit: it would be better to check the exception message/error code to distinguish between the really expected `IllegalStateException` vs something else (this is quite generic exception)

Reply:
I think this is targeted enough here to be fine.
We often do not have messages in IllegalArgumentExceptions, if the `checkState` is meant as the equivalent of an always active assert statement. This is mainly for tracing error root causes in case of legitimate bugs, and the stack trace defines the exception even better than any error message.